python/ccf/ledger.py

```diff
@@ -595,14 +606,29 @@ def __init__(self, filename: str):
         super().__init__(filename)
         self._filename = filename
         self._file_size = os.path.getsize(filename)
-        super()._read_header()
 
-    def commit_seqno(self):
-        try:
-            return int(self._filename.split("committed_")[1])
-        except IndexError:
-            # Snapshot is not yet committed
-            return None
+        entry_start_pos = super()._read_header()
+
+        # Snapshots embed evidence receipt since 2.x
+        if self.is_committed() and not self.is_snapshot_file_1_x():
+            receipt_pos = entry_start_pos + self._header.size
+            receipt_bytes = _peek(self._file, num_bytes=None, pos=receipt_pos)
+
+            receipt = json.loads(receipt_bytes.decode("utf-8"))
+            root = ccf.receipt.root(receipt["leaf"], receipt["proof"])
+            node_cert = load_pem_x509_certificate(
+                receipt["cert"].encode(), default_backend()
+            )
+            ccf.receipt.verify(root, receipt["signature"], node_cert)
```

Comment:
I may be missing something here, but it looks like we're claiming to verify this snapshot but not actually checking that the receipt is linked to the snapshot? Is there a link here that I'm missing, or would this pass if we embedded an arbitrary (valid!) receipt in the snapshot file? Is there somewhere else that the Python infra confirms that this receipt covers the evidence of this snapshot?

Reply:
Good point. This was raised but not documented in this ticket or elsewhere. This effectively requires public custom claims and will be addressed in a follow-up PR. See #3031